src/main/java/gyro/azure/CloudBlobFileBackend.java

```diff
@@ -0,0 +1,91 @@
+package gyro.azure;
+
+import com.microsoft.azure.storage.CloudStorageAccount;
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import gyro.azure.storage.StorageAccountResource;
+import gyro.core.FileBackend;
+import gyro.core.GyroException;
+import gyro.core.Type;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.stream.Stream;
+
+@Type("cloudblob")
+public class CloudBlobFileBackend extends FileBackend {
+    private String storageAccount;
+    private String cloudBlobContainer;
+    private String prefix;
+
+    public String getStorageAccount() {
+        return storageAccount;
+    }
+
+    public void setStorageAccount(String storageAccount) {
+        this.storageAccount = storageAccount;
+    }
+
+    public String getCloudBlobContainer() {
+        return cloudBlobContainer;
+    }
+
+    public void setCloudBlobContainer(String cloudBlobContainer) {
+        this.cloudBlobContainer = cloudBlobContainer;
+    }
+
+    public String getPrefix() {
+        return prefix;
+    }
+
+    public void setPrefix(String prefix) {
+        this.prefix = prefix;
+    }
+
+    @Override
+    public Stream<String> list() throws Exception {
+        return null;
```

Comment:
Not sure what the right thing to do here, empty list or null. As-is I think this would prevent this implementation from being used to save state.

Reply:
LocalFileBackend (https://github.com/perfectsense/gyro/blob/master/core/src/main/java/gyro/core/LocalFileBackend.java#L42) returns an empty list, while S3FileBackend (https://github.com/perfectsense/gyro-aws-provider/blob/master/src/main/java/gyro/aws/S3FileBackend.java#L41) does not.
I guess we should return empty list for all of them if that prevents an issue down the line if want to save something to the state.

Reply:
I think `S3FileBackend` is incorrect too. :) And `LocalFileBackend` doesn't always return an empty stream. It only does that if the directory doesn't exist.

Reply:
Right, so we would want the list implementation for S3 and CloudBlob to return an empty stream rather than a null?

Reply:
Well, it should technically return any files in the container underneath prefix.